typescript/sdk/src/deploy-cctp-adapter.ts

```diff
@@ -0,0 +1,110 @@
+import { ethers } from 'ethers';
+import yargs from 'yargs';
+
+import { types, utils } from '@hyperlane-xyz/utils';
+
+import {
+  ChainMap, // ChainName,
+  Chains,
+  MultiProvider,
+  chainMetadata,
+} from '../src';
+
+import { hyperlaneContractAddresses } from './consts/environments';
+import {
+  AdapterType,
+  CctpAdapterConfig,
+  CctpAdapterDeployer,
+} from './middleware/liquidity-layer-v2/CctpAdapterDeployer';
+
+const circleDomainMapping = [
+  { hyperlaneDomain: chainMetadata[Chains.goerli].chainId, circleDomain: 0 },
+  { hyperlaneDomain: chainMetadata[Chains.fuji].chainId, circleDomain: 1 },
+];
+
+function buildCctpAdapterConfigMap(owner: types.Address) {
+  const config: ChainMap<CctpAdapterConfig> = {};
+  config[Chains.goerli] = {
+    type: AdapterType.CCTP,
+    tokenMessengerAddress: '0xd0c3da58f55358142b8d3e06c1c30c5c6114efe8',
+    token: '0x07865c6e87b9f70255377e024ace6630c1eaa37f',
+    tokenSymbol: 'USDC',
+    gasAmount: 300000,
+    circleDomainMapping: circleDomainMapping,
+    mailbox: hyperlaneContractAddresses.goerli.mailbox,
+    interchainGasPaymaster:
+      hyperlaneContractAddresses.goerli.defaultIsmInterchainGasPaymaster,
```

Comment:
This is quite the nit, but since you will not be using the default ISM here, you should probably use the "inner" IGP that does not apply the overhead of multisig ISM (and then at the router, you specify a gas amount that does include the mailbox overhead)

Reply:
Good point. Done.
Btw, do you guys have a criteria for much gas should be added for padding? Our transactions differ a little depending on the network they're relayed on, but was thinking of adding about 10% as padding (250,000 for max 222,000). How does that sound?

Reply:
I think @yorhodes last time checked the current mailbox overhead, but adding some generous padding to start with makes sense, and we can always adjust more accurately later